lisa/test_runner/tests/test_lisarunner.py

```diff
@@ -0,0 +1,411 @@
+import asyncio
+from typing import Optional
+from unittest.case import TestCase
+
+from lisa import schema
+from lisa.environment import load_environments
+from lisa.test_runner.lisarunner import LisaRunner
+from lisa.tests import test_platform, test_testsuite
+from lisa.tests.test_environment import generate_runbook as generate_env_runbook
+from lisa.tests.test_platform import deleted_envs, deployed_envs, prepared_envs
+from lisa.tests.test_testsuite import (
+    cleanup_cases_metadata,
+    generate_cases_metadata,
+    generate_cases_result,
+)
+from lisa.testsuite import TestStatus, simple_requirement
+from lisa.util import constants
+
+
+def generate_lisarunner(
```

Comment:
Correct me if I'm wrong, but the framework changes (while they looked expansive) were mostly just moving and renaming, and this test code is the majority of the new code.

Reply:
Yes. Recorded it in commit description. Moved EnvironmentSpace out from schema, as it's not used as a schema.